web/components/auth-context.tsx

```diff
@@ -41,6 +41,7 @@ export function AuthProvider({ children }: any) {
   useEffect(() => {
     return onIdTokenChanged(auth, async (fbUser) => {
       if (fbUser) {
+        setAuthCookies(await fbUser.getIdToken(), fbUser.refreshToken)
```

Comment:
When you call `firebaseLogin`, it doesn't resolve until this event handler runs, but it doesn't wait for this event handler to also resolve. So now that we want to do stuff right way after `firebaseLogin` -- like reload and ask for new server side props based on our new cookies -- we ought to make sure we set the cookies right away.